toolchain/ltrace.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+dist = distro.detect()
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop
+        a library function check_repo() should be implemented.
+        """
+
+        os_deps = ['git', 'wget', 'rpm', 'autoconf',
+                   'automake', 'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs = os_deps + sles_deps
+
+        elif dist.name == 'redhat':
+            rhel_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                         'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs = os_deps + rhel_deps
+
+        elif dist.name == 'ubuntu':
+            ubuntu_deps = ['elfutils', 'libelf-dev', 'libtool',
+                           'libelf1', 'librpmbuild3', 'binutils-dev']
+            pkgs = os_deps + ubuntu_deps
+        else:
+            print "Unsupported OS!"
```

Comment:
Also in case this branch is used, the pkgs is never set. I'd suggest using `pkgs` from the beginning and extend it of the dist-dependant versions by `pkgs += rhel_deps` and similarly for others.